amr-wind/core/field_ops.H

```diff
@@ -67,6 +67,79 @@ add(T1& dst, const T2& src, int srccomp, int dstcomp, int numcomp, int nghost)
     add(dst, src, srccomp, dstcomp, numcomp, amrex::IntVect(nghost));
 }
 
+/** Divide two fields \f$y = y / x\f$
+ *  \ingroup field_ops
+ *
+ *  \tparam T1 Field or ScratchField
+ *  \tparam T2 Field or ScratchField
+ *  \param [out] dst Field that is updated
+ *  \param [in] src Field to be divisor
+ *  \param [in] srccomp Starting component index of source field
+ *  \param [in] dstcomp Starting component index of destination field
+ *  \param [in] ncomp_src Number of components to divide by
+ *  \param [in] ncomp_dst Number of components to be divided
+ *  \param [in] nghost Number of ghost cells to be updated
+ *  \param [in] iter_src_comp Flag of whether src components should be iterated
+ */
+template <typename T1, typename T2>
+inline void divide(
+    T1& dst,
+    const T2& src,
+    const int srccomp,
+    const int dstcomp,
+    const int ncomp_src,
+    const int ncomp_dst,
+    const amrex::IntVect& nghost)
+{
+    const int nlevels = dst.repo().num_active_levels();
+    for (int lev = 0; lev < nlevels; ++lev) {
+        if (ncomp_dst == ncomp_src) {
+            amrex::MultiFab::Divide(
+                dst(lev), src(lev), srccomp, dstcomp, ncomp_dst, nghost);
+        } else if (ncomp_src == 1) {
+            for (int n = 0; n < ncomp_dst; ++n) {
+                amrex::MultiFab::Divide(
+                    dst(lev), src(lev), srccomp, dstcomp + n, ncomp_src,
+                    nghost);
+            }
+        } else {
+            amrex::Abort(
+                "field_ops::divide : number of divisor components (ncomp_src) "
+                "must either be 1 or match number of dividend components "
+                "(ncomp_dst).\n");
```

Comment:
I don't think you want the `.\n`. Abort adds punctuation for us

Reply:
```suggestion
                "(ncomp_dst).");
```